src/notification.cpp

```diff
@@ -311,28 +362,79 @@ class NotificationPrivate : public NotificationData
             onClicked: console.log("Clicked")
             onClosed: console.log("Closed, reason: " + reason)
         }
-        text: "Application notification, ID " + notification.replacesId
+        text: "Application notification" + (notification.replacesId ? " ID:" + notification.replacesId : "")
         onClicked: notification.publish()
     }
     \endqml
 
-    An example category definition file
-    /usr/share/lipstick/notificationcategories/x-nemo.example.conf:
+    In this case, the notification includes a specification for
+    'remote actions', which are D-Bus commands that the notification
+    manager may permit the user to invoke. (Note: Lipstick currently
+    supports invoking only the remote action named "default"). When
+    an action is invoked on the notification, the corresponding D-Bus
+    command is formulated and invoked, which allows the application
+    to be launched to handled the notification action, if required.
+ */
 
-    \code
-    x-nemo-icon=icon-lock-sms
-    x-nemo-preview-icon=icon-s-status-sms
-    x-nemo-feedback=sms
-    x-nemo-priority=70
-    x-nemo-user-removable=true
-    x-nemo-user-closeable=false
-    \endcode
+#ifdef Q_QDOC
+namespace libnemonotifications {
+#endif
+
+/*!
+    \namespace libnemonotifications
+ */
+/*!
+    \class libnemonotifications::Notification
+    \brief Allows notifications to be published
+
+    The Notification class is a convenience class for using notifications
+    based on the
+    \l {https://people.gnome.org/~mccann/docs/notification-spec/notification-spec-latest.html}
+    {Desktop Notifications Specification} as implemented in Nemo.
+
+    This class allows clients to create instances of notifications, which
+    can be used to communicate to the Lipstick Notification Manager via
+    D-Bus.  This simplifies the process of creating, listing and closing
+    notifications, since the necessary communications are handled by the
+    class.
+
+    Note: although \c Notification is shown here in the \c libnemonotification namespace, it is
+    in fact defined in the global namespace.  The namespace here is used to work around limitations
+    of the qdoc documentation generator.
```

Comment:
What do you think would be a useful comment?  I think I would add:
'qdoc fails to resolve references when the project contains a QML type and a C++ type with the same name.'
Does adding that help?